kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/enumeration/DragAndDropHelper.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.workbench.common.dmn.client.editors.types.listview.constraint.enumeration;
+
+import java.util.Objects;
+
+import elemental2.dom.Element;
+import elemental2.dom.Event;
+import elemental2.dom.HTMLElement;
+import elemental2.dom.MouseEvent;
+import elemental2.dom.NodeList;
+import org.kie.workbench.common.stunner.core.util.StringUtils;
+
+class DragAndDropHelper {
+
+    private final HTMLElement dragArea;
+    private final HTMLElement addButtonContainer;
+    private int clickedYPosition;
+    private int startYPosition;
+    private HTMLElement draggingElement;
+
+    static final int ITEM_HEIGHT = 40;
+    static final String DRAGGABLE_ITEM_CLASS = ".draggable";
+
+    public DragAndDropHelper(final HTMLElement dragArea,
+                             final HTMLElement addButtonContainer) {
+
+        this.dragArea = dragArea;
+        this.addButtonContainer = addButtonContainer;
+        init();
+    }
+
+    void init() {
```

Comment:
I think this method could be `private`:
```suggestion
    private void init() {
```